DatadogRUM/Tests/Integrations/TelemetryReceiverTests.swift

```diff
@@ -404,10 +404,20 @@ class TelemetryReceiverTests: XCTestCase {
         XCTAssertEqual(event?.service, "dd-sdk-ios")
         XCTAssertEqual(event?.source, .reactNative)
         XCTAssertEqual(event?.telemetry.message, "[Mobile Metric] \(randomName)")
-        DDAssertReflectionEqual(event?.telemetry.telemetryInfo, randomAttributes)
+        randomAttributes.forEach { key, value in
+            DDAssertReflectionEqual(event?.telemetry.telemetryInfo[key], value)
+        }
+        let device = try XCTUnwrap(event?.telemetry.device)
+        XCTAssertTrue(device.model?.isEmpty == false)
+        XCTAssertTrue(device.brand?.isEmpty == false)
+        XCTAssertTrue(device.architecture?.isEmpty == false)
+        let os = try XCTUnwrap(event?.telemetry.os)
+        XCTAssertTrue(os.version?.isEmpty == false)
+        XCTAssertTrue(os.name?.isEmpty == false)
+        XCTAssertTrue(os.build?.isEmpty == false)
```

Comment:
suggestion/ We can improve this test to be more specific by mocking some values for `device` and `os` through `featureScope.contextMock`.